### PR TITLE
Fixed Black screen upon load custom screen and Added clear custom scene button

### PR DESCRIPTION
**Pull request by mohan-manthri**

### CODE REVIEW

File: lib/landing_screen.dart [hunk 1]
```diff
@@ -218,12 +218,46 @@ class _LandingScreenState extends State<LandingScreen> with Positioning {
...
+        }
+      },
+    );
+  }
+
+  void resetCustomScene(CustomMapStyleSettings customMapStyleSettings) {
+    customMapStyleSettings.reset();
+    _hereMapController.mapScene.loadSceneForMapScheme(
+      MapScheme.normalDay,
+      (_) {},
```
Comment (ppiatkowski-here) [inline lib/landing_screen.dart:244]: I think we should check for errors in the callback and at least log something if there is an error.

File: lib/landing_screen.dart [hunk 2]
```diff
@@ -218,12 +218,48 @@ class _LandingScreenState extends State<LandingScreen> with Positioning {
...
+      },
+    );
+  }
+
+  void resetCustomScene(CustomMapStyleSettings customMapStyleSettings) {
+    customMapStyleSettings.reset();
+    _hereMapController.mapScene.loadSceneForMapScheme(
+      MapScheme.normalDay,
+      (MapError? error) {
+        print('Map scene not loaded. MapError: ${error?.toString()}');
```
Comment (ppiatkowski-here) [inline lib/landing_screen.dart:245]: This will log error if map is loaded successfully. `Map scene not loaded. MapError: null`

**Review by ppiatkowski-here — APPROVED** [review]

**Review by tstaszak89 — APPROVED** [review]

File: lib/common/custom_map_style_settings.dart [hunk 3]
```diff
@@ -66,4 +41,9 @@ class CustomMapStyleSettings extends ChangeNotifier {
     }
     return _customMapStyleFilepath!.split('/').last;
   }
+
+  void reset() {
+    _customMapStyleFilepath = null;
```
Comment (tstaszak89) [inline lib/common/custom_map_style_settings.dart:46]: Please also delete scenes directory in this situation

File: lib/common/file_utils.dart [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ * All rights reserved.
+ *
+ * This software and other materials contain proprietary information
+ * controlled by HERE and are protected by applicable copyright legislation.
+ * Any use and utilization of this software and other materials and
+ * disclosure to any third parties is conditional upon having a separate
+ * agreement with HERE for the access, use, utilization or disclosure of this
+ * software. In the absence of such agreement, the use of the software is not
+ * allowed.
+ */
```
Comment (tstaszak89) [inline lib/common/file_utils.dart:12]: Please use the same copyright header as it is used in rest of ref app source files

File: lib/common/custom_map_style_settings.dart (hunk 3 above)
Comment (mohan-manthri) [reply lib/common/custom_map_style_settings.dart:46]: @tstaszak89  directory deletion now moved to separate and handled in landing screen

Comment (tstaszak89) [reply lib/common/custom_map_style_settings.dart:46]: Ok

File: lib/common/file_utils.dart [hunk 5]
```diff
@@ -0,0 +1,56 @@
+/*
```
Comment (tstaszak89) [inline lib/common/file_utils.dart:1]: There is already file_utility.dart file, so please move content of this file into file_utility.dart

**Review by ppiatkowski-here — APPROVED** [review]

**Review by tstaszak89 — CHANGES_REQUESTED** [review]

**Review by tstaszak89 — APPROVED** [review]

File: lib/common/file_utils.dart [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import 'dart:io';
+
+import 'package:path_provider/path_provider.dart';
+
+const scene_directory_name = 'scenes';
```
[the remark this answers is not in the document]
Comment (mohan-manthri) [reply lib/common/file_utils.dart]: I updated this

**Review by ppiatkowski-here — APPROVED** [review]

**Pull request merged**